loader/linux/icd_linux.c

```diff
@@ -61,76 +62,74 @@ void khrIcdOsVendorsEnumerate(void)
-                    fileName = malloc(strlen(dirEntry->d_name) + strlen(vendorPath) + 2);
-                    if (!fileName)
-                    {
-                        KHR_ICD_TRACE("Failed allocate space for ICD file path\n");
-                        break;
-                    }
-                    sprintf(fileName, "%s/%s", vendorPath, dirEntry->d_name);
-
-                    // open the file and read its contents
-                    fin = fopen(fileName, "r");
-                    if (!fin)
-                    {
-                        free(fileName);
-                        break;
-                    }
-                    fseek(fin, 0, SEEK_END);
-                    bufferSize = ftell(fin);
-
-                    buffer = malloc(bufferSize+1);
-                    if (!buffer)
-                    {
-                        free(fileName);
-                        fclose(fin);
-                        break;
-                    }
-                    memset(buffer, 0, bufferSize+1);
-                    fseek(fin, 0, SEEK_SET);
-                    if (bufferSize != (long)fread(buffer, 1, bufferSize, fin) )
-                    {
-                        free(fileName);
-                        free(buffer);
-                        fclose(fin);
-                        break;
-                    }
-                    // ignore a newline at the end of the file
-                    if (buffer[bufferSize-1] == '\n') buffer[bufferSize-1] = '\0';
-
-                    // load the string read from the file
-                    khrIcdVendorAdd(buffer);
+                    continue;
+                }
+                if (strcmp(dirEntry->d_name + strlen(dirEntry->d_name) - strlen(extension), extension) )
+                {
+                    continue;
+                }
+
+                // allocate space for the full path of the vendor library name
+                fileName = malloc(strlen(dirEntry->d_name) + strlen(vendorPath) + 2);
+                if (!fileName)
+                {
+                    KHR_ICD_TRACE("Failed allocate space for ICD file path\n");
+                    continue;
+                }
+                sprintf(fileName, "%s/%s", vendorPath, dirEntry->d_name);
 
+                // open the file and read its contents
+                fin = fopen(fileName, "r");
+                if (!fin)
+                {
+                    free(fileName);
+                    continue;
+                }
+                fseek(fin, 0, SEEK_END);
+                bufferSize = ftell(fin);
+
+                buffer = malloc(bufferSize+1);
+                if (!buffer)
+                {
+                    free(fileName);
+                    fclose(fin);
+                    continue;
+                }
+                memset(buffer, 0, bufferSize+1);
+                fseek(fin, 0, SEEK_SET);
+                if (bufferSize != (long)fread(buffer, 1, bufferSize, fin) )
```

Comment:
```suggestion
                if ((long)fread(buffer, 1, bufferSize, fin)  != bufferSize)
```
to avoid also some inattention bug the day we type `=` instead of `!=` or `==` :-)